src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -174,4 +175,28 @@ public static Lab parseLab(String labNumber) throws ParseException {
         return new Lab(trimmedLab);
     }
 
+    /**
+     * Parses a {@code String labStatus} into a {@code LabStatus}.
+     * Leading and trailing whitespaces will be trimmed.
+     */
+    public static LabStatus parseLabStatus(String labStatus) throws ParseException {
+        requireNonNull(labStatus);
+        String trimmedLabStatus = labStatus.trim();
+
+        if (!LabStatus.isValidLabStatus(trimmedLabStatus)) {
+            throw new ParseException(LabStatus.MESSAGE_CONSTRAINTS);
+        }
+
+        switch(trimmedLabStatus) {
+        case "u":
+            return LabStatus.UNSUBMITTED;
+        case "s":
+            return LabStatus.SUBMITTED;
+        case "g":
+            return LabStatus.GRADED;
+        default:
+            throw new IllegalArgumentException("Did not expect " + trimmedLabStatus);
```

Comment:
Might it be better to put the `throw new ParseException` here? Not sure if the method calling this catches the `IllegalArgumentException`

Reply:
That makes sense, thanks for catching the issue